In [2]:
#import necessary libraries 
import numpy as np
import scipy.stats as stats

In [3]:
#define sample data
sample_data = [1.13,1.55,1.43,0.92,1.25,1.36,0.85,1.97,1.48,1.20,1.33,1.18,1.22,1.29]

In [4]:
#calculate sample mean
sample_mean = np.mean(sample_data)

In [5]:
sample_mean

1.297142857142857

In [6]:
#calculate sample standard deviation
sample_std = np.std(sample_data, ddof=1)

In [7]:
sample_std

0.2737609448863277

In [8]:
#calculate sample size
n = len(sample_data)

In [9]:
n

14

In [10]:
#calculate standard error
standard_error = sample_std / np.sqrt(n)

In [13]:
# Find the critical t-value for a 99% confidence interval
# Since we're using a sample and small scale data, we use the t-distribution.
confidence_level = 0.99
alpha = 1 - confidence_level
t_critical = stats.t.ppf(1 - alpha/2, df=n-1)

In [14]:
#calculate margin of error
margin_of_error = t_critical * standard_error

In [15]:
#Calculate confidence interval
confidence_interval = (sample_mean - margin_of_error, sample_mean + margin_of_error)

In [16]:
print(f"Sample Mean: {sample_mean}")
print(f"Margin of Error: {margin_of_error}")
print(f"99% Confidence Interval: {confidence_interval}")

Sample Mean: 1.297142857142857
Margin of Error: 0.22039524052651474
99% Confidence Interval: (1.076747616616342, 1.5175380976693718)


Explanation :- We calculated a 99% confidence interval. The sample mean is 1.2971, with a margin of error of 0.2204. This results in a confidence interval of [1.0767,1.5175]. We are 99% confident that  data lies within this range."

In [17]:
#define population standard deviation
polpulation_std = 0.2


In [22]:
# Find the critical z-value for a 99% confidence interval
confidence_level = 0.99
alpha = 1 - confidence_level
z_critical = stats.norm.ppf(1 - alpha/2)  # z-distribution critical value



In [23]:
# Calculate margin of error
margin_of_error = z_critical * standard_error

In [25]:
#Calculate confidence interval
confidence_interval = (sample_mean - margin_of_error, sample_mean + margin_of_error)
print(f"Sample Mean: {sample_mean}")
print(f"Margin of Error: {margin_of_error}")
print(f"99% Confidence Interval: {confidence_interval}")

Sample Mean: 1.297142857142857
Margin of Error: 0.1884623286188768
99% Confidence Interval: (1.1086805285239802, 1.4856051857617336)


Summary :- We calculated a 99% confidence interval for the population standard deviation based on the sample data. The sample mean is 1.2971, with a margin of error of 0.1885. This gives a confidence interval of [1.1087,1.4856]. We are 99% confident that the true population standard deviation lies within this range."